### ISSUE TITLE
IfcDiff TypeError: Object of type PrettyOrderedSet is not JSON serializable 

### ISSUE THREAD

**Issue by PeterLawrence** [IfcOpenShell/IfcOpenShell#4259]
While parsing the differences in properties between two IFC files IFCDiff reports the following

```
File "D:\Python311\Lib\json\encoder.py", line 180, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type PrettyOrderedSet is not JSON serializable
```
This is using the command line: ifcdiff.py old.ifc new.ifc -r property

This error occurs at line 239 of IfcDiff.py in the function export, which is.
```
            json.dump(
                {
                    "added": list(self.added_elements),
                    "deleted": list(self.deleted_elements),
                    "changed": self.change_register,
                },
                diff_file,
                indent=4,
            )
```
As a work around,  I've added a new function to the class IfcDiff

```
    def set_default(self,obj):
        if isinstance(obj, set):
            return list(obj)
        return str(obj)
```
and modifying the json.dump call as followed..

```
            json.dump(
                {
                    "added": list(self.added_elements),
                    "deleted": list(self.deleted_elements),
                    "changed": self.change_register,
                },
                diff_file,
                indent=4,
                default=self.set_default
            )
```
The above changes resolved the issue.

On further examination the PrettyOrderedSet, which caused this issue contained the following strings
```
[root['Pset_DoorCommon']['Type']]
[root['Pset_SpaceOccupancyRequirements']]
[root['Pset_RampCommon']['DirectionOfTravel'], root['Pset_RampCommon']['Base Level'], root['Pset_RampCommon']['Type'], root['Pset_RampCommon']['Export to IFC As'], root['Pset_RampCommon']['Width'], root['Pset_RampCommon']['Type IFC Predefined Type'], root['Pset_RampCommon']['RequiredSlope'], root['Pset_RampCommon']['IfcGUID']]
```

So the  issue seems to be with the processing Pset_DoorCommon and Pset_RampCommon data in the two ifc files.

**Comment by Andrej730** [comment]
@PeterLawrence hi! can you please share example .ifc files to reproduce the issue?

**Comment by Andrej730** [comment]
I've managed to reproduce this by running ifcdiff between two files - one containing `Pset_WindowCommon.Status` and other not. It seems IfcDiff is returning `PrettyOrderedSet` to store the data.
Example files.
[new.ifc.txt](https://github.com/IfcOpenShell/IfcOpenShell/files/14096425/new.ifc.txt)
[old.ifc.txt](https://github.com/IfcOpenShell/IfcOpenShell/files/14096426/old.ifc.txt)

I've added `json.dump` to fallback to lists in case it will meet `OrderedSet` or `set`.

> As a work around, I've added a new function to the class IfcDiff
> ```python
>     def set_default(self,obj):
>         if isinstance(obj, set):
>             return list(obj)
>         return str(obj)
>  ```

I think this workaround wouldn't work since the problem is with serializing ordered sets, not just sets (`isinstance(OrderedSet, set) == False`).
